controllers/topology/current_state.go

```diff
@@ -18,12 +18,132 @@ package topology
 
 import (
 	"context"
+	"fmt"
 
-	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/cluster-api/api/v1alpha4"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// Gets the current state of the Cluster topology.
-func (r *ClusterReconciler) getCurrentState(ctx context.Context, cluster *clusterv1.Cluster) (*clusterTopologyState, error) {
-	// TODO: add get class logic; also remove nolint exception from clusterTopologyState and machineDeploymentTopologyState
-	return nil, nil
+// getCurrentState gets information about the current state of a Cluster by inspecting the state of the InfrastructureCluster,
+// the ControlPlane, and the MachineDeployments associated with the Cluster.
+func (r *ClusterReconciler) getCurrentState(ctx context.Context, cluster *v1alpha4.Cluster, class v1alpha4.ClusterClass) (*clusterTopologyState, error) {
```

Comment:
We usually import the api/v1alpha4  package as clusterv1 (I assume to avoid having to change all occurances with the next version). Would be great to keep this consistent.

Reply:
👍